compiler/qsc_eval/src/backend.rs

```diff
@@ -419,6 +420,36 @@ impl Backend for SparseSim {
                 self.apply_noise(q);
                 Some(Ok(Value::unit()))
             }
+            "Apply" => {
```

Comment:
Did you check if Apply needs to be handled in other backends? I had to add handling for other intrinsics in a few places.

Reply:
I did. In this case, circuits already do the right thing in that they treat any custom intrinsic as a box with the params displayed:
![image](https://github.com/user-attachments/assets/a4548483-8851-407c-88c4-6347effc74a3)
For the other backends/scenarios they all will fail with an error about an unrecognized intrinsic which is actually the desired behavior. Neither QIR generation nor resource estimation know how to handle an arbitrary unitary, so should fail if invoked on code that includes this call.